meshtastic/interfaces/ble/discovery.py

```
@@ -524,7 +562,30 @@ def _discard_cached_client() -> None:
+                ) = None
+                with self._client_lock:
+                    if self._client is client:
+                        discarded_client = self._client
+                        self._client = None
+                if discarded_client is not None:
+                    _close_discovery_client_best_effort(discarded_client)
+                raise DiscoveryClientError.invalid_client(
+                    resolved_factory,
+                    type(client),
+                    ["discover", "_discover"],
+                )
```

Ruff (0.15.5)

[warning] Abstract `raise` to an inner function

(TRY301)